src/transactions.ts

```diff
@@ -19,12 +19,14 @@ import type * as db from './db/mod';
  * [[Replicache.subscribe]] and allows read operations on the
  * database.
  */
-export interface ReadTransaction {
+export interface ReadTransaction<
+  Value extends ReadonlyJSONValue = ReadonlyJSONValue,
+> {
   /**
    * Get a single value from the database. If the `key` is not present this
    * returns `undefined`.
    */
-  get(key: string): Promise<JSONValue | undefined>;
+  get(key: string): Promise<Value | undefined>;
```

Comment:
This is kind of ugly. Maybe it is better to not have WriteTransaction extend ReadTransaction any more